src/dbnode/storage/series/series.go

```diff
@@ -301,6 +301,20 @@ func (s *dbSeries) ReadEncoded(
 	return r, err
 }
 
+func (s *dbSeries) FetchBlocksForColdFlush(
+	ctx context.Context,
+	start time.Time,
+	version int,
+	nsCtx namespace.Context,
+) ([]xio.BlockReader, error) {
+	// This needs a write lock because the version on underlying buckets need
+	// to be modified.
+	s.Lock()
```

Comment:
This is going to be called in a tight loop right? Would be good to avoid the defer overhead here if possible

Reply:
```golang
s.Lock()
br, err := s.buffer.FetchBlocksForColdFlush(ctx, start, version, nsCtx)
s.Unlock()

return br, err
```